pwiz_tools/Skyline/Util/BioMassCalc.cs

```diff
@@ -598,7 +598,8 @@ public string StripLabelsFromFormula(string desc)
             }
             var parse = desc;
             var dictAtomCounts = new Dictionary<string, int>();

```

Comment:
I think you should use the class "OrderedDictionary" here.
That class is similar to Java's "LinkedHashMap": It's a dictionary, but it remembers the order that you put things into it. I think that would enable you to get rid of the "atomOrder" argument below.

Reply:
Turns out OrderedDictionary is pretty dissimilar from Dictionary or SortedDictionary, to the point where using it is more complicated that it's worth due to code reuse issues. I think the atomOrder stuff needs to stay. It was worth investigating though, thanks.

Reply:
> Turns out OrderedDictionary is pretty dissimilar from Dictionary or SortedDictionary, to the point where using it is more complicated that it's worth due to code reuse issues. I think the atomOrder stuff needs to stay. It was worth investigating though, thanks.

Ohhhh. Yes, I see that "OrderedDictionary" is not a generic class. That is unfortunate.